util/src/test/java/org/killbill/billing/util/bcd/TestBillCycleDayCalculator.java

```diff
@@ -172,7 +172,7 @@ public void testAlignProposedNextBillCycleDate1() {
         // BCD > proposed day of the month
```

Comment:
Could we add a few unit tests here to cover for non monthly billing periods - at least the one matching the scenario?